File: ShoppingV6Java/app/src/main/java/dk/itu/shopping/UIFragment.java
package dk.itu.shopping;

import android.content.res.Configuration;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

public class UIFragment extends Fragment {
    //GUI variables
    Button addItem, listItems;
    private TextView newWhat, newWhere;

    // Model: Database of items
    ShoppingViewModel viewModel;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        super.onCreateView(inflater, container, savedInstanceState);
        final View v = inflater.inflate(R.layout.fragment_ui, container, false);

        //Text Fields
        newWhat = v.findViewById(R.id.what_text);
        newWhere = v.findViewById(R.id.where_text);

        //Buttons
        listItems = v.findViewById(R.id.list);
        addItem = v.findViewById(R.id.add_button);

        viewModel = new ViewModelProvider(requireActivity()).get(ShoppingViewModel.class);

        //Only show List button in Portraitmode
        if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT) {
            listItems.setOnClickListener(view ->
                    getActivity().
                            getSupportFragmentManager()
                            .beginTransaction()
                            .replace(R.id.main_fragment,
                                    new ListFragment()).commit());
        }

        // adding a new thing
        addItem.setOnClickListener(view -> viewModel.onAddItemClick(newWhat, newWhere, getActivity()));

        return v;
    }
}

File: ShoppingV5Kotlin/app/src/main/java/dk/itu/shoppingv5/UIFragment.kt
package dk.itu.shoppingv5

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider


class UIFragment : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        val v = inflater.inflate(R.layout.fragment_ui, container, false)
        val viewModel = ViewModelProvider(requireActivity())[ShoppingViewModel::class.java]

        //Text Fields
        val itemWhat: TextView = v.findViewById(R.id.what_text)
        val itemWhere: TextView = v.findViewById(R.id.where_text)

        //Buttons
        val addItem: Button = v.findViewById(R.id.add_button)
        val findItems: Button = v.findViewById(R.id.find_button)
        val deleteItem: Button = v.findViewById(R.id.delete_button)

        // adding a new thing
        addItem.setOnClickListener {
            activity?.let { fragmentActivity ->
                viewModel.onAddItemClick(
                    itemWhat,
                    itemWhere,
                    fragmentActivity
                )
            }
        }

        // deleting a thing
        deleteItem.setOnClickListener {
            activity?.let { fragmentActivity ->
                viewModel.onDeleteItemClick(itemWhat, fragmentActivity)
            }
        }

        // finding a thing
        findItems.setOnClickListener {
            viewModel.onFindItemClick(itemWhat, itemWhere)
        }

        return v
    }
}